qdrant_client/local/local_collection.py

```diff
@@ -666,6 +685,117 @@ def recommend_groups(
             with_lookup_collection=with_lookup_collection,
         )
 
+    def _preprocess_discover(
+        self,
+        target: Optional[types.RecommendExample] = None,
+        context: Optional[Sequence[types.ContextExamplePair]] = None,
+        query_filter: Optional[types.Filter] = None,
+        using: Optional[str] = None,
+        lookup_from_collection: Optional["LocalCollection"] = None,
+        lookup_from_vector_name: Optional[str] = None,
+    ) -> Tuple[Optional[List[float]], List[ContextPair], types.Filter]:
+        collection = lookup_from_collection if lookup_from_collection is not None else self
+        search_in_vector_name = using if using is not None else DEFAULT_VECTOR_NAME
+        vector_name = (
+            lookup_from_vector_name
+            if lookup_from_vector_name is not None
+            else search_in_vector_name
+        )
+        context = context if context is not None else []
+
+        # Validate inputs
+        if target is None and len(context) == 0:
+            raise ValueError("No target or context given")
+
+        # Turn every example into vectors
+        target_vector: Optional[types.NumpyArray] = None
```

Comment:
I think you can just define a type hint here without assigning `None`
also it seems like it should be `Optional[List[float]]`
and `target_vector = collection.vectors[vector_name][idx]` should be `target_vector = collection.vectors[vector_name][idx].tolist()` 

it does not really break anything because of further conversions, just a nit one

Reply:
addressed